src/crawlee/_request.py

```diff
@@ -25,6 +25,53 @@
 from crawlee._utils.urls import extract_query_params, validate_http_url
 
 
+def _normalize_headers(headers: Mapping[str, str]) -> dict[str, str]:
+    """Converts all header keys to lowercase and returns them sorted by key."""
+    normalized_headers = {k.lower(): v for k, v in headers.items()}
+    sorted_headers = sorted(normalized_headers.items())
+    return dict(sorted_headers)
+
+
+class HttpHeaders(BaseModel, Mapping[str, str]):
```

Comment:
Maybe we could move this to utils, types or something?

Reply:
I will leave it in the `_types` then.